controllers/platform/services.go

```diff
@@ -58,30 +55,32 @@ func (action *serviceAction) Handle(ctx context.Context, platform *operatorapi.S
 		return nil, err
 	}
 
-	if err := createDataIndexComponents(ctx, action.client, platform); err != nil {
-		return nil, err
+	if platform.Spec.Services.DataIndex != nil {
```

Comment:
Should we get into this block if `platform.Spec.Services.DataIndex.Enabled == false`?

Reply:
it's correct as it is. the `Enabled` field does not control whether a service is deployed. instead, it controls whether its used by the "prod" workflows -
https://github.com/apache/incubator-kie-kogito-serverless-operator/blob/a96add5c995496ef2d02f313114efed7d74a4725/api/v1alpha08/sonataflowplatform_services_types.go#L27-L29

Reply:
Thank you for the reference.
In which case should this be set to false, assuming the workflow's profile is set to `prod`?

Reply:
no, this flag only affects "prod" workflows. if it's a dev workflow, this flag won't even be checked by the operator. when a user elects to set `platform.spec.services.dataIndex.enabled` to `true` (which is the default), they are saying they want "prod" workflows to use this data index.

Reply:
I agree, once we have configs changed, we then rollout every workflow removing this property so then we scale it to 0. For this initial version, is fine the way it is.

Reply:
@ricardozanini it's worth asking the question though, to @masayag's point... is this a use-case we should be supporting anyway? What IS the use-case? When would an admin or dev need/want to "disable" these services? FWIW, the user could just as easily set the service replicas to 0 themselves. Is this "enabled" field necessary?

I'm good either way... just asking the Q.

Reply:
I keep asking these to myself. Let's reevaluate the need for it. I remember we discussed how to have the platform deploy the services and that's what we came up with. If we enable deployment, we should also disable it. We must have a response to irt or change the structure in a way that "disable" won't make sense.

Reply:
Maybe this was discussed before, so dropping some ideas here:

Perhaps the desired behavior here is more aligned with [k8s rolloutStrategy](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.26/#deploymentstrategy-v1-apps). 
I'd explore the idea of replacing the `Enabled` flag with a `rolloutStrategy` property.
It leaves more room for introducing behavior types on CR change. In case of a change should be reflected in the workflows, the `rolloutStrategy` will determine the behavior. Few values to `rolloutStrategy`:
* Disabled - do nothing when platform CR is changed. this requires some out-of-sync between platform configuration to what is currently served by the workflows.
* Recreate - restart all with new configuration
* RollingUpdate - rolling the changes.
* Detached - detaching the service from workflow definition.

I'm also okay with not having it at all, but making the sync-status with the platform configuration visible is useful for debugging.
For once, I'd go with a single flag or strategy for all of the services and not per service.
Also, perhaps the future will knative service will not need this type of handling at all.

Reply:
@masayag that sounds reasonable, mind opening an issue here describing this behavior so we can discuss it more in-depth? We might have to implement it preferably before the cluster scope one. @tchughesiv wdyt about his proposal?

Reply:
I also think this all sounds reasonable, with `RollingUpdate` as the default when this new "flag" is not set. @ricardozanini I'm not sure the cluster-scoped changes would need to wait for this though... what what your thinking with that?

Reply:
Won't we replicate the spec in the cluster scope? I'd rather have the change once and propagate to the cluster-scoped requirements.

Reply:
no, we will not replicate the spec in cluster-scope. we previously discussed this and decided to only reference the chosen platform in spec and leave it at that. the referenced SonataFlowPlatform will be the single source of truth.

Reply:
See https://github.com/apache/incubator-kie-kogito-serverless-operator/issues/321